hikey_debian.mk

```diff
@@ -0,0 +1,408 @@
+
+linux-defconfig: $(LINUX_PATH)/.config
+
+LINUX_COMMON_FLAGS += ARCH=arm64 deb-pkg LOCALVERSION=-optee-rpb
+UPSTREAM_KERNEL := $(if $(wildcard $(LINUX_PATH)/arch/arm64/boot/dts/hisilicon/hi6220-hikey.dts),1,0)
+ifeq ($(UPSTREAM_KERNEL),0)
+LINUX_COMMON_FLAGS += hi6220-hikey.dtb
+DTB = $(LINUX_PATH)/arch/arm64/boot/dts/hi6220-hikey.dtb
+else
+LINUX_COMMON_FLAGS += hisilicon/hi6220-hikey.dtb
+DTB = $(LINUX_PATH)/arch/arm64/boot/dts/hisilicon/hi6220-hikey.dtb
+endif
+
+linux: linux-common
+
+.PHONY: linux-defconfig-clean
+linux-defconfig-clean: linux-defconfig-clean-common
+
+LINUX_CLEAN_COMMON_FLAGS += ARCH=arm64
+
+.PHONY: linux-clean
+linux-clean: linux-clean-common
+
+LINUX_CLEANER_COMMON_FLAGS += ARCH=arm64
+
+.PHONY: linux-cleaner
+linux-cleaner: linux-cleaner-common
+
+################################################################################
+# OP-TEE
+################################################################################
+OPTEE_OS_COMMON_FLAGS += PLATFORM=hikey CFG_TEE_TA_LOG_LEVEL=3 CFG_CONSOLE_UART=$(CFG_SW_CONSOLE_UART)
+OPTEE_OS_CLEAN_COMMON_FLAGS += PLATFORM=hikey
+
+optee-os: optee-os-common
+
+.PHONY: optee-os-clean
+optee-os-clean: optee-os-clean-common
+
+optee-client: optee-client-common
+
+.PHONY: optee-client-clean
+optee-client-clean: optee-client-clean-common
+
+################################################################################
+# xtest / optee_test
+################################################################################
+
+xtest: xtest-common
+
+# FIXME:
+# "make clean" in xtest: fails if optee_os has been cleaned previously
+.PHONY: xtest-clean
+xtest-clean: xtest-clean-common
+	rm -rf $(OPTEE_TEST_OUT_PATH)
+
+.PHONY: xtest-patch
+xtest-patch: xtest-patch-common
+
+################################################################################
+# aes-pef
+################################################################################
+PERF_FLAGS := CROSS_COMPILE_HOST=$(CROSS_COMPILE_NS_USER) \
+	CROSS_COMPILE_TA=$(CROSS_COMPILE_S_USER) \
+	TA_DEV_KIT_DIR=$(OPTEE_OS_TA_DEV_KIT_DIR)
+
+aes-perf: optee-os optee-client
+	$(MAKE) -C -j$(NPROC) $(AESPERF_PATH) $(PERF_FLAGS)
+
+.PHONY: aes-perf-clean
+aes-perf-clean:
+	rm -rf $(AESPERF_PATH)/out
+
+################################################################################
+# sha-perf
+################################################################################
+sha-perf: optee-os optee-client
+	$(MAKE) -C -j$(NPROC) $(SHAPERF_PATH) $(PERF_FLAGS)
+
+.PHONY: sha-perf-clean
+sha-perf-clean:
+	rm -rf $(SHAPERF_PATH)/out
+
+################################################################################
+# Boot Image
+################################################################################
+.PHONY: boot-img
+boot-img:
+ifeq ("$(wildcard $(BOOT_IMG))","")
+	echo "Downloading Debian HiKey boot image ..."
+	wget https://builds.96boards.org/snapshots/hikey/linaro/debian/410/boot-fat.uefi.img.gz -O $(BOOT_IMG).gz
```

Comment:
I guess we have to match this with the system img below which doesn't have a 'dynamic' link due to the date in the filename?